packages/aws-cdk-lib/aws-appsync/lib/source-api-association.ts

```diff
@@ -0,0 +1,220 @@
+import { Construct } from 'constructs';
+import { CfnSourceApiAssociation } from './appsync.generated';
+import { IGraphqlApi } from './graphqlapi-base';
+import { Effect, IRole, PolicyStatement } from '../../aws-iam';
+import { Fn, IResource, Lazy, Resource } from '../../core';
+
+/**
+ * Merge type used to associate the source API
+ */
+export enum MergeType {
+  /**
+  * Manual merge. The merge must be triggered manually when the source API has changed.
+  */
+  MANUAL_MERGE = 'MANUAL_MERGE',
+
+  /**
+  * Auto merge. The merge is triggered automatically when the source API has changed.
+  */
+  AUTO_MERGE = 'AUTO_MERGE',
+}
+
+/**
+ * Interface for AppSync Source Api Association
+ */
+export interface ISourceApiAssociation extends IResource {
+
+  /**
+   * The association id.
+   */
+  readonly associationId: string;
+
+  /**
+   * The association arn.
+   */
+  readonly associationArn: string;
+
+  /**
+   * The source api in the association.
+   */
+  readonly sourceApi: IGraphqlApi;
+
+  /**
+   * The merged api in the association.
+   */
+  readonly mergedApi: IGraphqlApi;
+}
+
+/**
+ * The attributes for imported AppSync Source Api Association.
+ */
+export interface SourceApiAssociationAttributes {
+  /**
+   * The association arn.
+   */
+  readonly associationArn: string;
+
+  /**
+   * The source api in the association.
+   */
+  readonly sourceApi: IGraphqlApi;
+
+  /**
+   * The merged api in the association.
+   */
+  readonly mergedApi: IGraphqlApi;
+}
+
+/**
+* Properties for SourceApiAssociation which associates an AppSync Source API with an AppSync Merged API
+*/
+export interface SourceApiAssociationProps {
+
+  /**
+   * The source api to associate.
+   */
+  readonly sourceApi: IGraphqlApi;
+
+  /**
+   * The merged api to associate.
+   */
+  readonly mergedApi: IGraphqlApi;
+
+  /**
+   * The merged api execution role for adding the access policy for the source api.
+   */
+  readonly mergedApiExecutionRole?: IRole;
+
+  /**
+   * The merge type for the source
+   *
+   * @default - AUTO_MERGE
+   */
+  readonly mergeType?: MergeType;
+
+  /**
+   * The description of the source api association
+   *
+   * @default - None
+   */
+  readonly description?: string;
+}
+
+/**
+ * AppSync SourceApiAssociation which associates an AppSync source API to an AppSync Merged API.
+ * The initial creation of the SourceApiAssociation merges the source API into the Merged API schema.
+ */
+export class SourceApiAssociation extends Resource implements ISourceApiAssociation {
+
+  /**
+   * Import Appsync Source Api Association from source API, merged api, and merge type.
+   */
+  public static fromSourceApiAssociationAttributes(scope: Construct, id: string, attrs: SourceApiAssociationAttributes): ISourceApiAssociation {
```

Comment:
I'm not necessarily against adding this function, but its use is very unlikely, right?

There isn't a single API that accepts an `ISourceApiAssociation` ?